### PR TITLE
Move `_KINETO_AVAILABLE` check to profiler

### LINKED ISSUE #11826: Move `_*_AVAILABLE` checks to their respective files.
## Proposed refactor

Move all the variables defined here that are only used in one file
https://github.com/PyTorchLightning/pytorch-lightning/blob/1e36cffbca3f92855d82907370544fade1656da6/pytorch_lightning/utilities/imports.py#L98-L134
to their respective files.

### Motivation

This encapsulates code and avoids clutter from the utilities.

### Pitch

Move:

https://github.com/Lightning-AI/lightning/blob/50fd12f841bc10ec093e80d0acf3e3b51cd927fe/src/pytorch_lightning/utilities/imports.py#L28-L51

To the files that use them.

The rest I don't think should be moved.
______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @borda

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #11826

It looks like the remaining items can't be moved as they are shared in several places. So we can probably close #11826

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
:books: Documentation preview :books:: https://pytorch-lightning--18575.org.readthedocs.build/en/18575/

cc @justusschock @awaelchli

### CODE REVIEW

File: src/lightning/pytorch/profilers/pytorch.py [hunk 1]
```diff
@@ -22,25 +22,23 @@
...
     from lightning.pytorch.core.module import LightningModule
 
-if _KINETO_AVAILABLE:
-    from torch.profiler import ProfilerAction, ProfilerActivity, tensorboard_trace_handler
 
 log = logging.getLogger(__name__)
 warning_cache = WarningCache()
 
 _PROFILER = Union[torch.profiler.profile, torch.autograd.profiler.profile, torch.autograd.profiler.emit_nvtx]
+_KINETO_AVAILABLE = torch.profiler.kineto_available()
```
Comment (awaelchli) [inline src/lightning/pytorch/profilers/pytorch.py:41]: This check only takes a few ns, but I could also inline it if you like.

**Review by Borda — APPROVED** [review]

**Review by carmocca — APPROVED** [review]
